### PR TITLE
[JEP-200] This should be a plugin, not a JAR library

### PR DESCRIPTION
**Pull request by jglick**
For https://github.com/jenkinsci/jenkins/pull/3120. Needs to be released and then used from feature plugins:

- [ ] `publish-over-cifs`
- [ ] `publish-over-dropbox`
- [ ] `publish-over-ftp`
- [ ] `publish-over-ssh`

@reviewbybees

### CODE REVIEW

**Comment by slide** [comment]
The Jenkinsfile should be updated as part of this to just be ```buildPlugin()``` then.

File: pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>publish-over</artifactId>
-    <packaging>jar</packaging>
+    <packaging>hpi</packaging>
     <name>Publish Over</name>
```
Comment (oleg-nenashev) [inline pom.xml]: Name needs to be changed. "Utilities for 'Publish Over' plugins" or so?

Comment (jglick) [reply pom.xml]: Maybe. Not a priority for me.

**Comment by jglick** [comment]
@slide what is your plan here? This library-cum-plugin needs to be released, then the four downstream plugins need to update their dependencies to the new release version, and themselves be released.

**Comment by jglick** [comment]
Please update [this page](https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200) with plugin release versions.

**Comment by oleg-nenashev** [comment]
I have updated the page and confirmed that the new plugin appears in the UC and gets installed correctly

**Comment by oleg-nenashev** [comment]
All dependent plugins have been released

**Pull request merged**